runtime/src/attestations.rs

```diff
@@ -118,16 +145,21 @@ impl<T: Trait> Module<T> {
 			let mut valid = Vec::new();
 			let invalid = Vec::new();
 
-			for (auth_index, _) in head.validator_indices
-				.iter()
-				.enumerate()
-				.filter(|(_, bit)| *bit)
 			{
-				let stash_id = validators.get(auth_index)
-					.expect("auth_index checked to be within bounds in `check_candidates`; qed")
-					.clone();
-
-				valid.push(stash_id);
+				let attesting_indices = head.validator_indices
+					.iter()
+					.enumerate()
+					.filter(|(_, bit)| *bit)
+					.inspect(|&(auth_index, _)| {
+						let stash_id = validators.get(auth_index)
+							.expect("auth_index checked to be within bounds in `check_candidates`; qed")
```

Comment:
hm, but `check_candidates` is not local, can we actually depend on this property here? Because everything that we can depend here is the interface of `Get` and it doesn't state anything about length of the returned vector.

As an aside, I think `Get` is supposed to be used with static values which do not change.

Reply:
It's unsafe to rely on `Get` not to change, since a runtime upgrade could change it. Honestly, having to write a new trait for every get-like operation is easily replaced by using type-functional traits like `Get`. I don't see why we _shouldn't_ do it.